File: dotnet/EntityFX.NetBenchamarks.Core/Generic/RandomMemoryBenchmark.cs
using System.Runtime.CompilerServices;
using System.Threading.Tasks;
using System;
namespace EntityFX.NetBenchmark.Core.Generic
{

    public class RandomMemoryBenchmark: RandomMemoryBenchmarkBase<MemoryBenchmarkResult>
    {
        public override MemoryBenchmarkResult BenchImplementation() 
        {
            return BenchRandomMemory();
        }

        public override BenchResult PopulateResult(BenchResult benchResult, MemoryBenchmarkResult result)
        {
            benchResult.Points = Convert.ToDecimal(result.Average * Ratio);
            benchResult.Result = Convert.ToDecimal(result.Average);
            benchResult.Units = "MB/s";
            benchResult.Output = result.Output;
            return benchResult;
        }

    }
}

File: src/lua/generic/randomMemoryBenchmark.lua
RandomMemoryBenchmark = class(RandomMemoryBenchmarkBase, function(a, writer, printToConsole)
    RandomMemoryBenchmarkBase.init(a, writer, printToConsole)
    a.name = "RandomMemoryBenchmark"
end)

function RandomMemoryBenchmark:benchImplementation()
    return self:benchRandomMemory()
end

function RandomMemoryBenchmark:populateResult(benchResult, memoryBenchmarkResult)
    benchResult.points = memoryBenchmarkResult.average * self.ratio
    benchResult.result = memoryBenchmarkResult.average
    benchResult.units = "MB/s"
    benchResult.output = memoryBenchmarkResult.output
    return benchResult;
end